state/upgrades.go

```diff
@@ -250,3 +253,97 @@ func SplitMigrationStatusMessages(pool *StatePool) error {
 	}
 	return st.runRawTransaction(ops)
 }
+
+func PopulateApplicationStorageUniqueID(pool *StatePool, getBrokerFunc func(model *Model) (caas.Broker, error)) error {
+	logger.Infof("[adis] inside PopulateApplicationStorageUniqueID")
+	st, err := pool.SystemState()
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	var ops []txn.Op
+
+	err = runForAllModelStates(pool, func(st *State) error {
+		model, err := st.Model()
+		if err != nil {
+			return errors.Trace(err)
+		}
+		logger.Infof("[adis] trying to upgrade applications for model %q", model.Name())
+
+		if model.Type() != ModelTypeCAAS {
+			logger.Infof("[adis] skipping because model %q is not a caas model", model.Name())
+			return nil
+		}
+
+		logger.Infof("[adis] model %q preparing for populating storage id", model.Name())
+
+		applications, closer := st.db().GetCollection(applicationsC)
+		defer closer()
+
+		iter := applications.Find(nil).Iter()
+		defer iter.Close()
+
+		broker, err := getBrokerFunc(model)
+		if err != nil {
+			return errors.Trace(err)
+		}
+
+		getAnnotation := func(appName string, mode string, includeClusterIP bool) (map[string]interface{}, error) {
+			service, err := broker.GetService(appName, caas.DeploymentMode(mode), includeClusterIP)
+			if err != nil {
+				return map[string]interface{}{}, err
+			}
+			logger.Infof("[adis] getAnnotation data is: %+v", service.Status.Data)
+			return service.Status.Data, nil
```

Comment:
????? We should not be messing with status data to get annotations off a k8s resource.